esmvalcore/_recipe.py

```diff
@@ -1074,8 +1075,19 @@ def _initialize_variables(self, raw_variable, raw_datasets):
                 activity = get_activity(variable)
                 if activity:
                     variable['activity'] = activity
-            check.variable(variable, required_keys)
-        variables = self._expand_ensemble(variables)
+            if 'sub_experiment' in variable:
+                subexperiment_keys = deepcopy(required_keys)
+                subexperiment_keys.update({'sub_experiment'})
+                try:
+                    subexperiment_keys.remove('start_year')
+                    subexperiment_keys.remove('end_year')
+                except KeyError:
+                    continue
```

Comment:
Why should be remove this? (Side note: [`discard`](https://docs.python.org/3/library/stdtypes.html#frozenset.discard) might be preferable to `remove`.)